packages/profiler/src/index.ts

```diff
@@ -51,7 +51,8 @@ export class Profiler {
       resolver,
       paths,
       base_path: basePath,
-      contracts_directory: contractsDirectory
+      contracts_directory: contractsDirectory,
+      compiler // { name, version }
```

Comment:
So I'm not sure we want to call it this, since we already have `truffle compile --compiler=<solc|vyper|external>`. This shouldn't cause problems, but it might be weird? Thoughts?

Reply:
Hm, tricky.  I agree it's not ideal to have overlap there. `currentCompiler`? `compilerSpecifier`?  IDK.

Reply:
`compilerSpecification`? definitely something other than compiler

Reply:
note that `compiler` is used in this form exactly in @truffle/db, so that's why I went with this

Reply:
Like, it goes in `options` like this there?

Reply:
Yep!

Reply:
Then I guess this is no worse than what already exists! :)